src/passes/OptimizeInstructions.cpp

```diff
@@ -3870,7 +3866,22 @@ struct OptimizeInstructions
       }
     }
     {
-      // fval(C) / -x   ==>  -C / x
+      // NaN - x   ==>   NaN'
+      // NaN / x   ==>   NaN'
+      Const* c;
+      if ((matches(curr, binary(Sub, fval(&c), pure())) ||
```

Comment:
We should move away from using `pure()` in such things, as it prevents optimization. Here even if pure we can return `getDroppedChildrenAndAppend()` or `getResultOfFirst` are better.

In this case, `getDroppedChildrenAndAppend(binary, .., binary->left)` will work. That command gets drops of all necessary children, which is either a drop of `x` or nothing (since we never need to keep a drop of a Const), and then finishes with returning the NaN.

Reply:
I agree. What about making a separate PR that replaces all pure with `getDroppedChildrenAndAppend`?